lib/src/working_copy.rs

```diff
@@ -154,10 +154,23 @@ fn create_parent_dirs(disk_path: &Path) {
         .unwrap_or_else(|_| panic!("failed to create parent directories for {:?}", &disk_path));
 }
 
+fn mtime_from_metadata(metadata: &Metadata) -> MillisSinceEpoch {
+    let time = metadata
+        .modified()
+        .expect("File mtime not supported on this platform?");
+    let since_epoch = time
+        .duration_since(UNIX_EPOCH)
+        .expect("mtime before unix epoch");
+    MillisSinceEpoch(
+        since_epoch
+            .as_millis()
+            .try_into()
```

Comment:
Rust question: what is this trying to go into? I guess whatever type MillisSinceEpoch() accepts?

Reply:
Exactly. That's a `u64`. I agree it's unclear. I switched to `u64::try_from()` to make it clearer.